pygmt/tests/test_grdgradient.py

```diff
@@ -28,29 +199,22 @@ def test_grdgradient_outgrid(grid):
         result = grdgradient(grid=grid, outgrid=tmpfile.name, azimuth=10, direction="c")
         assert result is None  # return value is None
         assert os.path.exists(path=tmpfile.name)  # check that outgrid exists
-        result = (
-            grdinfo(grid=tmpfile.name, force_scan="a", per_column="n").strip().split()
-        )
-    npt.assert_allclose(float(result[4]), -0.0045060496)  # min
-    npt.assert_allclose(float(result[5]), 0.0575332976)  # max
-    # Check spherically weighted statistics below
-    npt.assert_allclose(float(result[10]), 0.000384754501283)  # median
-    npt.assert_allclose(float(result[12]), 0.00285958005568)  # mean
+        temp_grid = load_dataarray(tmpfile.name)
+        xr.testing.assert_allclose(a=temp_grid, b=expected_grid)
 
 
-def test_grdgradient_no_outgrid(grid):
+def test_grdgradient_no_outgrid(grid, expected_grid):
     """
     Test the azimuth and direction parameters for grdgradient with no set
     outgrid.
     """
-    temp_grid = grdgradient(grid=grid, azimuth=10, direction="c")
-    assert temp_grid.dims == ("lat", "lon")
-    assert temp_grid.gmt.gtype == 1  # Geographic grid
-    assert temp_grid.gmt.registration == 1  # Pixel registration
-    npt.assert_allclose(temp_grid.min(), -0.0045060496)
-    npt.assert_allclose(temp_grid.max(), 0.0575332976)
-    npt.assert_allclose(temp_grid.median(), 0.0004889865522272885)
-    npt.assert_allclose(temp_grid.mean(), 0.0028633063193410635)
+    result = grdgradient(grid=grid, azimuth=10, direction="c")
```

Comment:
In future GMT releases, azimuth and direction will not be allowed together because the two options are contradictory (see discussion in https://github.com/GenericMappingTools/gmt/pull/6327). Sorry for the extra work, but I recommend updating this test to use only one of these two parameters.